torch/nn/modules/module.py

```diff
@@ -586,87 +613,162 @@ def state_dict(self, destination=None, prefix='', keep_vars=False):
         """
         if destination is None:
             destination = OrderedDict()
+            destination._versions = OrderedDict()
+        if hasattr(destination, '_versions'):
```

Comment:
Why would it ever not have this attribute here? We should really make all of those args private, they're not meant to be used by others